core/src/allocator/rust.rs

```diff
@@ -35,6 +35,38 @@ fn round_size(size: usize) -> usize {
 pub struct RustAllocator;
 
 unsafe impl Allocator for RustAllocator {
+    fn calloc(&mut self, count: usize, size: usize) -> *mut u8 {
+        if count == 0 || size == 0 {
+            return ptr::null_mut();
+        }
+
+        let total_size = count.checked_mul(size).expect("overflow");
+
+        let total_size = round_size(total_size);
+
+        // Calculate the total allocated size including header
+        let alloc_size = HEADER_SIZE + total_size;
+
+        let layout = if let Ok(layout) = Layout::from_size_align(alloc_size, ALLOC_ALIGN) {
+            layout
+        } else {
+            return ptr::null_mut();
+        };
+
+        let ptr = unsafe { alloc::alloc(layout) };
+
+        if ptr.is_null() {
+            return ptr::null_mut();
+        }
+
+        let header = unsafe { &mut *(ptr as *mut Header) };
+        header.size = total_size;
+
+        let ptr = unsafe { ptr.add(HEADER_SIZE) };
+        unsafe { std::ptr::write_bytes(ptr, 0, total_size) };
```

Comment:
Instead of allocating and then setting the bytes to zero you can also call `alloc::alloc_zeroed`, that can be sometimes faster.